clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java

```diff
@@ -0,0 +1,245 @@
+   */
+  @Override
+  public HttpClientBuilder fromConfig(Function<String, String> configuration) {
+    String uri = configuration.apply(CONF_NESSIE_URI);
+    if (uri == null) {
+      uri = configuration.apply(CONF_NESSIE_URL);
+    }
+    if (uri != null) {
+      this.uri = URI.create(uri);
+    }
+    String username = configuration.apply(CONF_NESSIE_USERNAME);
+    if (username != null) {
+      this.username = username;
+    }
+    String password = configuration.apply(CONF_NESSIE_PASSWORD);
+    if (password != null) {
+      this.password = password;
+    }
+    String authType = configuration.apply(CONF_NESSIE_AUTH_TYPE);
+    if (authType != null) {
+      this.authType = AuthType.valueOf(authType);
+    }
+    String tracing = configuration.apply(CONF_NESSIE_TRACING);
+    if (tracing != null) {
+      this.tracing = Boolean.parseBoolean(tracing);
+    }
+    String owner = configuration.apply(CONF_NESSIE_OWNER);
+    if (owner != null) {
+      Validation.validateOwner(owner);
+      this.owner = owner;
+    }
+    String repo = configuration.apply(CONF_NESSIE_REPOSITORY);
+    if (repo != null) {
+      Validation.validateRepo(repo);
+      this.repo = repo;
+    }
+    return this;
+  }
+
+  /**
+   * Set the authentication type. Default is {@link AuthType#NONE}.
+   *
+   * @param authType new auth-type
+   * @return {@code this}
+   */
+  @Override
+  public HttpClientBuilder withAuthType(AuthType authType) {
+    this.authType = authType;
+    return this;
+  }
+
+  /**
+   * Set the Nessie server URI. A server URI must be configured.
+   *
+   * @param uri server URI
+   * @return {@code this}
+   */
+  @Override
+  public HttpClientBuilder withUri(URI uri) {
+    this.uri = uri;
+    return this;
+  }
+
+  /**
+   * Convenience method for {@link #withUri(URI)} taking a string.
+   *
+   * @param uri server URI
+   * @return {@code this}
+   */
+  @Override
+  public HttpClientBuilder withUri(String uri) {
+    return withUri(URI.create(uri));
+  }
+
+  /**
+   * Set the repository owner + ID within the Nessie instance.
+   *
+   * <p>{@link NessieClient#getContentsApi()} and {@link NessieClient#getTreeApi()} are scoped to
+   * the repository specified by this option.
+   *
+   * @param owner repository owner
+   * @param repo repository ID
+   * @return {@code this}
+   */
+  public HttpClientBuilder withRepoOwner(String owner, String repo) {
+    Validation.validateOwner(owner);
+    this.owner = owner;
+    this.repo = repo;
+    return this;
+  }
+
+  /**
+   * Set the username for {@link AuthType#BASIC} authentication.
+   *
+   * @param username username
+   * @return {@code this}
+   */
+  public HttpClientBuilder withUsername(String username) {
+    this.username = username;
+    return this;
+  }
+
+  /**
+   * Set the password for {@link AuthType#BASIC} authentication.
+   *
+   * @param password password
+   * @return {@code this}
+   */
+  public HttpClientBuilder withPassword(String password) {
```

Comment:
Would it make sense to move auth parameters to the base `NessieClientBuilder` interface? I imagine we ought to support the same set of credential types across all clients (even though the credential transport may have to be different).

Reply:
Right, those should not really differ semantically. Added that.